pennylane/_qubit_device.py

```diff
@@ -427,8 +420,16 @@ def sample_basis_states(self, number_of_states, state_probability):
         Returns:
             List[int]: the sampled basis states
         """
+        if self.shots is None:
+            raise DeprecationWarning(
+                "In future, the number of shots have to be explicitly set on the device "
+                "when using sample-based measurements."
+            )
```

Comment:
I use a warning here to have some backwards compatibility. But in future we may want to turn this into an error.

Reply:
The reason why I am hesitant is that this would break _any_ code that ever used sampling without explicitly stating the number of shots. I feel this is an extremely invasive change!

With these changes, the former behaviour is replicated to some extent: the unassuming user will just get 1000 samples.

I am not sure if there is a better way to do this?

Reply:
Thanks for the clarification. I think this works well for now then, and then we can later decide to change it to an error. This way, it'll be backwards compatible, and since we've used 1000 shots as default before, it makes sense to have it here as well.